src/net.cpp

```diff
@@ -92,7 +93,7 @@ bool fListen = true;
 bool fRelayTxes = true;
 CCriticalSection cs_mapLocalHost;
 std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(cs_mapLocalHost);
-static bool vfLimited[NET_MAX] GUARDED_BY(cs_mapLocalHost) = {};
+static std::bitset<NET_MAX> g_nets_reachable GUARDED_BY(cs_mapLocalHost)(std::bitset<NET_MAX>().set());
```

Comment:
nit, I still have a preference for
```cpp
static auto g_nets_reachable GUARDED_BY(cs_mapLocalHost){std::bitset<NET_MAX>().set()};
```
since it doesn't repeat the type — here the type is obvious.

Reply:
It's a small difference, but I included it because I think it makes the declaration more readable - after GUARDED_BY seems a bit buried if you're scanning down the page.

Reply:
This is the first use of `std::bitset` in this project. It's a *small* set, the memory savings here are minimal compared to the extra code that is being introduced to handle it.
I think the original code was fine.

Reply:
Yeah - I think of `bitset` as just `vector<bool>` without the interface requirements of `vector`, thus able to be more constrained (e.g. fixed size). But I'll revert to the array to narrow the focus of this PR.